modAionImpl/src/org/aion/zero/impl/db/AionBlockStore.java

```diff
@@ -1178,12 +1159,12 @@ public void close() {
         lock.writeLock().lock();
 
         try {
-            indexDS.close();
+            index.close();
         } catch (Exception e) {
             LOG.error("Not able to close the index database:", e);
         } finally {
             try {
-                blocksDS.close();
+                blocks.close();
```

Comment:
Does anything bad happen if index failed to close, but blocks gets closed?

Reply:
If either one fails to close it may lead to data corruption, but most of the time nothing bad happens. There are methods in place to recover from data corruption. In a bad case, the user may need to revert back a few blocks. In the worst case, they may need to wipe the database. There is functionality for correcting the index database, so that might be recovered. If the block database is broken, they need to re-sync from Genesis.